### ISSUE TITLE
Upload to Asset Store step fails when run as external job

### ISSUE THREAD

**Issue by DmitriyYukhanov** [RockTomate/Issues#5]
**Steps to Reproduce**
* create and configure Upload to Asset Store step
* put it to Unity 2017.4.1 project
* configure Run External Job step to run this step, like this:
![image](https://user-images.githubusercontent.com/883587/78681194-156ce480-78f5-11ea-9636-3ecaffb0f5ec.png)
* run job
* see Run External Job fails

**Expected Results**
I expected to see asset uploaded to the Asset Store successfully

**Actual Results**
Upload fails with this in logs:
```
Authorization Required
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
HardCodeLab.RockTomate.Steps.AssetStorePublisher.<OnLoginCompleted>c__AnonStorey1:<>m__0(HttpWebResponse, String) (at Assets\!Service\RockTomate\Scripts\Steps\Classes\AssetStorePublisher\AssetStoreAPI.cs:117)
HardCodeLab.RockTomate.Steps.AssetStorePublisher.<HandleRequest>c__AnonStorey6:<>m__0() (at Assets\!Service\RockTomate\Scripts\Steps\Classes\AssetStorePublisher\AssetStoreAPI.cs:332)
UnityEditor.EditorApplication:Internal_CallDelayFunctions() (at C:\buildslave\unity\build\Editor\Mono\EditorApplication.cs:136)
 
(Filename: Assets/!Service/RockTomate/Scripts/Steps/Classes/AssetStorePublisher/AssetStoreAPI.cs Line: 117)

Request timed out while processing request "https://public-cdn.cloud.unity3d.com/config/production", HTTP error code 0
.[UnityConnect] Request timed out.
Refresh: detecting if any assets need to be imported or removed ... Refresh: elapses 0.019315 seconds (Nothing changed)
Connect to CacheServer localhost:64959
Updating ProjectSettings/ProjectSettings.asset - GUID: 00000000000000004000000000000000...
 done. [Time: 3.420400 ms] 
Disconnect from CacheServer
Refreshing native plugins compatible for Editor in 3.06 ms, found 4 plugins.
Preloading 2 native plugins for Editor in 0.17 ms.

----- Total AssetImport time: 0.085522s, AssetImport time: 0.030815s, Asset hashing: 0.000000s [0 B, 0.000000 mb/s]
----- Cache Server detail: CacheServerIntegrate time: 0.000000s, CacheServer download: 0.000036s [0 B, 0.000000 mb/s]
----- Cache Server detail: Total assets requested: 1, Cached Assets: 0, Not available: 0, Not supported: 1.
----- Cache Server detail: CacheMove: 0.000000s, CacheDelete: 0.000000s.

Refresh: detecting if any assets need to be imported or removed ... Refresh: elapses 0.019331 seconds (Nothing changed)
[Package Manager] Server::Kill -- Server was shutdown
Cleanup mono
```

**Comment by DmitriyYukhanov** [comment]
Here is a view from the 2017.41 project where upload to the store job is located and executed:
![image](https://user-images.githubusercontent.com/883587/78683232-b5c40880-78f7-11ea-82c7-6910a7f90e80.png)

**Comment by ElmarTalibzade** [comment]
Could you double-check the credentials?

**Comment by DmitriyYukhanov** [comment]
Hey, I've re-checked and it looks correct.
One possibly important detail: username has + character (like a+b@gmail.com)

**Comment by ElmarTalibzade** [comment]
In UploadToAssetStoreStep.cs, place a breakpoint on line 94. It should be something like this: `_assetStoreApi.Login(Username, Password);`

Then, attach it to unity debugger and run the step. 

Could you confirm that `Username` and `Password` variables that are passed into the method are correct?

**Comment by DmitriyYukhanov** [comment]
Yes I do confirm variables contain correct login and password.
But it still returns Unauthorized http error =(

I did double-checked credentials once again and logged into asset store using them - they do work for 100%. I used them previously with another automation uploading tool.

![image](https://user-images.githubusercontent.com/883587/79209367-fc5fb880-7e4b-11ea-8229-6a3ae2001e5f.png)

**Comment by DmitriyYukhanov** [comment]
Alright, I just tested it with other user and pass and it works!
It's nothing to do with your code, it's on Unity's API side looks like since I did tested it with postman now.

Super weird thing is both old (not working) and new (I just created and tested) are working fine in built-in login window of the Asset Store Tools.

I'm closing this case and will choose another login without + character since I believe it could cause some glitches.

Thank you for taking your time looking into this!
